In [7]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt


file_path = r"E:\Projects\Gamezone Orders Data\gamezone_orders_data_cleaned.csv"
df = pd.read_csv(
    file_path,
    parse_dates=['purchase_ts_cleaned', 'ship_ts'],
    dtype={
        'purchase_year': 'Int64',
        'purchase_month': 'Int64',
        'time_to_ship': 'Int64',
        'revenue': 'float'
    },
    encoding='utf-8',
    keep_default_na=False,
    na_values=['']  # Only treat empty strings as NaN
)

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21864 entries, 0 to 21863
Data columns (total 21 columns):
 #   Column                           Non-Null Count  Dtype         
---  ------                           --------------  -----         
 0   user_id                          21864 non-null  object        
 1   order_id                         21864 non-null  object        
 2   purchase_ts                      21864 non-null  object        
 3   purchase_ts_cleaned              21863 non-null  datetime64[ns]
 4   purchase_year                    21864 non-null  Int64         
 5   purchase_month                   21864 non-null  Int64         
 6   time_to_ship                     21864 non-null  Int64         
 7   ship_ts                          21864 non-null  datetime64[ns]
 8   product_name                     21864 non-null  object        
 9   product_name_cleaned             21864 non-null  object        
 10  product_id                       21864 non-null  object   

In [4]:
df['country_code'].nunique()

151

# 🔍 Data Validation Report - Gamezone Orders

**Dataset:** orders.csv  
**Date Validated:** 2025-11-06  
**Analyst:** [Your Name]  
**Total Rows:** 21,864  
**Date Range:** [Your start date] to [Your end date]

---

## 📊 Executive Summary

| Metric | Value |
|--------|-------|
| **Total Records** | 21,864 |
| **Valid Records (Clean)** | 19,864 (90.5%) |
| **Records with Issues** | 2,000 (9.5%) |
| **Total Issues Identified** | 4 |
| **Critical Issues** | 1 |
| **Resolved Issues** | 12 |
| **Unresolvable Issues** | 1 |

---

## 🎯 Validation Outcome

✅ **DATASET IS READY FOR ANALYSIS** with documented caveats

⚠️ One critical issue (Invalid Ship Dates) requires conditional exclusion in operational analyses

---

## 🚨 Critical Issue Summary

### Issue #4: Invalid Ship Dates
- **Severity:** 🔴 HIGH  
- **Impact:** 2,000 orders (9.5%)  
- **Root Cause:** Data corruption - ship dates 1-247 days before purchase dates  
- **Solvability:** ❌ Unresolvable (no ground truth available)  
- **Resolution:** Flagged with `has_invalid_shipdate=True`
- **Treatment:**
  - ✅ **INCLUDED** in revenue, customer, product, marketing analyses  
  - ❌ **EXCLUDED** from operational shipping metrics
- **Documentation:** See Issue #4 details below + `Data_Cleaning_Master.ipynb`

---

## ✅ Minor Issues (All Resolved)

| Issue | Description | Rows | % | Status |
|-------|-------------|------|---|--------|
| **#1** | User ID auto-converted to scientific notation | 36 | 0.16% | ✅ Resolved - Converted to text |
| **#2** | Inconsistent date formats | 10 | 0.05% | ✅ Resolved - Parsed with dayfirst=True |
| **#3** | Incomplete data (missing purchase_ts) | 1 | 0.00% | ✅ Resolved - Negligible impact, retained |

---
